tests/Aspire.Azure.AI.OpenAI.Tests/ConformanceTests.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.ConformanceTests;
+using Azure.AI.OpenAI;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Azure.AI.OpenAI.Tests;
+
+public class ConformanceTests : ConformanceTests<OpenAIClient, AzureOpenAISettings>
+{
+    // Fake connection string for cases when credentials are unavailable and need to switch to raw connection string
+    protected const string Endpoint = "https://aspireopenaitests.openai.azure.com/";
+    protected const string Key = "fake";
+
+    protected override ServiceLifetime ServiceLifetime => ServiceLifetime.Singleton;
+
+    protected override string[] RequiredLogCategories => new string[] {
+        "Azure.Core",
+        // Not triggered because no connection is made
+        // "Azure.AI.OpenAI",
+    };
+
+    protected override bool SupportsKeyedRegistrations => true;
+
+    protected override string JsonSchemaPath => "src/Components/Aspire.Azure.AI.OpenAI/ConfigurationSchema.json";
+
+    protected override string ValidJsonConfig => """
+        {
+          "Aspire": {
+            "Azure": {
+              "AI": {
+                "OpenAI": {
+                  "Endpoint": "http://YOUR_URI",
+                  "Tracing": true,
+                  "ClientOptions": {
+                    "ConnectionIdleTimeout": "PT1S",
+                    "EnableCrossEntityTransactions": true,
+                    "RetryOptions": {
+                      "Mode": "Fixed",
+                      "MaxDelay": "PT3S"  
+                    },
+                    "TransportType": "AmqpWebSockets"
+                  }
+                }
+              }
+            }
+          }
+        }
+        """;
+
+    protected override (string json, string error)[] InvalidJsonToErrorMessage => new[]
+        {
+            ("""{"Aspire": { "Azure": { "AI":{ "OpenAI": {"Endpoint": "YOUR_URI"}}}}}""", "Value does not match format \"uri\""),
+            ("""{"Aspire": { "Azure": { "AI":{ "OpenAI": {"Endpoint": "http://YOUR_URI", "Tracing": "false"}}}}}""", "Value is \"string\" but should be \"boolean\""),
+        };
+
+    protected override string ActivitySourceName => throw new NotImplementedException();
+
+    // When credentials are not available, we switch to using raw connection string (otherwise we get CredentialUnavailableException)
+    protected KeyValuePair<string, string?> GetMainConfigEntry(string? key)
```

Comment:
Who calls this? Is it just a copy paste from ServiceBus?

Reply:
Removed, now I see it was just for SB